src/main/java/zabi/minecraft/nbttooltip/parse_engine/BWHumanReadableParser.java

```diff
@@ -51,39 +49,37 @@ private void addCompoundToTooltip(List<Text> tooltip, Tag base, String pad, Stri
 		});
 	}
 	
-	private void addListToTooltip(List<Text> tooltip, Tag base, String pad, String padIncrement, boolean splitLongStrings) {
-		AbstractListTag<?> tag = (AbstractListTag<?>) base;
+	private void addListToTooltip(List<Text> tooltip, NbtElement base, String pad, String padIncrement, boolean splitLongStrings) {
+		AbstractNbtList<?> tag = (AbstractNbtList<?>) base;
 		int index = 0;
-		Iterator<? extends Tag> iter = tag.iterator();
-		while (iter.hasNext()) {
-			Tag nbtnext = iter.next();
-			if (nbtnext instanceof AbstractListTag || nbtnext instanceof CompoundTag) {
-				tooltip.add(new LiteralText(pad + "["+index+"]: {"));
-				unwrapTag(tooltip, nbtnext, pad+padIncrement, "", padIncrement, splitLongStrings);
-				tooltip.add(new LiteralText(pad+"}"));
+		for (NbtElement nbtnext : tag) {
+			if (nbtnext instanceof AbstractNbtList || nbtnext instanceof NbtCompound) {
+				tooltip.add(new LiteralText(pad + "[" + index + "]: {"));
+				unwrapTag(tooltip, nbtnext, pad + padIncrement, "", padIncrement, splitLongStrings);
+				tooltip.add(new LiteralText(pad + "}"));
 			} else {
-				addValueToTooltip(tooltip, nbtnext, "["+index+"]", pad, splitLongStrings);
+				addValueToTooltip(tooltip, nbtnext, "[" + index + "]", pad, splitLongStrings);
 			}
 			index++;
 		}
 	}
 	
-	private static void addValueToTooltip(List<Text> tooltip, Tag nbt, String name, String pad, boolean splitLongStrings) {
+	private static void addValueToTooltip(List<Text> tooltip, NbtElement nbt, String name, String pad, boolean splitLongStrings) {
 		String toBeAdded = nbt.toString();
 		if (!splitLongStrings || toBeAdded.length() < line_split_threshold) {
-			tooltip.add(new LiteralText(pad+name+": "+nbt.toString()));
+			tooltip.add(new LiteralText(pad+name+": "+ nbt));
 		} else {
 			int added = 0;
 			tooltip.add(new LiteralText(pad+name+":"));
 			while (added < toBeAdded.length()) {
 				int nextChunk = Math.min(line_split_threshold, toBeAdded.length() - added);
-				StringBuilder sb = new StringBuilder(Formatting.AQUA.toString())
-					.append("|")
-					.append(Formatting.RESET.toString())
-					.append(pad)
-					.append("   ")
-					.append(toBeAdded.substring(added, added+nextChunk));
-				tooltip.add(new LiteralText(sb.toString()));
+				String sb = Formatting.AQUA +
```

Comment:
String concat chains allocate more objects than necessary. Since this might deal with a lot of strings very often, StringBuilders are the better choice here